src/index.ts

```diff
@@ -34,73 +38,31 @@ export async function execute(filenames: string[], options: Partial<ExecutionOpt
     const executionOptions = Object.assign(defaultExecutionOptions, options);
 
     let manifest = await PP.getManifest(executionOptions.root);
+    let componentDefinitions = await getComponentDefinitionMap(executionOptions.root);
+
+    let pathFormatter = (component: ComponentDefinition) => {
+        if (component.scripts.length < 1) return;
+        component.scripts = component.scripts.map((script: ComponentScript) => {
+            script.uri = path.join(
+                options.root ? options.root : __dirname,
+                new URL(script.uri).pathname
+            );
+            return script;
+        });
+    };
 
-    let nodeDefs = await getComponentDefinitionMap(executionOptions.root);
-
-    // wait for all files to be read, lexed, and parsed, but don't exit on the first error
-    let parsedFiles = await pSettle(
-        filenames.map(async filename => {
-            let contents;
-            try {
-                contents = await readFile(filename, "utf-8");
-            } catch (err) {
-                return Promise.reject({
-                    message: `brs: can't open file '${filename}': [Errno ${err.errno}]`,
-                });
-            }
-
-            let lexer = new Lexer();
-            let preprocessor = new PP.Preprocessor();
-            let parser = new Parser();
-            [lexer, preprocessor, parser].forEach(emitter => emitter.onError(logError));
-
-            let scanResults = lexer.scan(contents, filename);
-            if (scanResults.errors.length > 0) {
-                return Promise.reject({
-                    message: "Error occurred during lexing",
-                });
-            }
-
-            let preprocessResults = preprocessor.preprocess(scanResults.tokens, manifest);
-            if (preprocessResults.errors.length > 0) {
-                return Promise.reject({
-                    message: "Error occurred during pre-processing",
-                });
-            }
-
-            let parseResults = parser.parse(preprocessResults.processedTokens);
-            if (parseResults.errors.length > 0) {
-                return Promise.reject({
-                    message: "Error occurred parsing",
-                });
-            }
-
-            return Promise.resolve(parseResults.statements);
-        })
+    let lexerParserFn = LexerParser.getLexerParserFn(manifest);
+    const interpreter = await Interpreter.withSubEnvsFromComponents(
+        componentDefinitions,
+        pathFormatter,
+        lexerParserFn
     );
-
-    // don't execute anything if there were reading, lexing, or parsing errors
-    if (parsedFiles.some(file => file.isRejected)) {
-        return Promise.reject({
-            messages: parsedFiles
-                .filter(file => file.isRejected)
-                .map(rejection => rejection.reason.message),
-        });
+    if (!interpreter) {
+        throw new Error("Unable to build interpreter with given component definitions.");
```

Comment:
Is this always a component definition error?

Reply:
It used to be the case, but after my last minor refactor this is no longer true. I'll update this to indicate a problem building the interpreter in general.